src/IO/S3/Credentials.cpp

```diff
@@ -21,6 +21,21 @@
 namespace DB::S3
 {
 
+namespace
+{
+
+bool areCredentialsEmptyOrExpired(const Aws::Auth::AWSCredentials & credentials, uint64_t expiration_window_seconds)
+{
+    if (credentials.IsEmpty())
+        return true;
+
```

Comment:
If my previous suggestion about adding `IsExpiredOrEmpty` above is ok, then here we could add
```
if (!expiration_window_seconds)
   return false;
```